File: 김은지/마법의-엘리베이터.js
function solution(storey) {
  const stoneCase = [];
  const stack = [[storey, 0]];

  while (stack.length > 0) {
    const [currNum, stone] = stack.pop();

    if (currNum < 10) {
      stoneCase.push(stone + Math.min(currNum, 10 - currNum + 1));
      continue;
    }

    const remainder = currNum % 10;
    const nextNum = Math.floor(currNum / 10);

    stack.push([nextNum, stone + remainder]);
    stack.push([nextNum + 1, stone + (10 - remainder)]);
  }

  return Math.min(...stoneCase);
}

File: 김현경/마법의_엘리베이터.py
# 1의 자리 숫자가 5보다 크면 올림, 작으면 내림, 5인 경우에는 10의 자리 숫자를 기준으로 선택

def solution(storey):
    answer = 0

    while (storey > 0):
        curr_num = storey % 10 # 1의 자리 수 추출
        
        if curr_num > 5: 
            answer += 10 - curr_num
            storey += 10 - curr_num

        elif curr_num < 5:
            answer += curr_num #버튼 누른 횟수 추가 
            storey -= curr_num
        else:
            if ((storey//10) % 10 >= 5):
                answer += 10 - curr_num
                storey += 10 - curr_num
            else:
                answer += curr_num 
                storey -= curr_num
        storey //= 10 # 현재 처리한 1의 자리 숫자를 제거 하고 다음 자리 수로 이동
        
    return answer